lib/lm_sensors.dart

```diff
@@ -0,0 +1,95 @@
+library dslink.system.lm_sensors;
+
+import "dart:async";
+import "dart:io";
+
+import "utils.dart" as util;
+
+final RegExp rawValuePattern = new RegExp(r"[+-.0-9]+");
+
+class SensorValue {
+  final double value;
+  final String unit;
+
+  SensorValue(this.value, [this.unit]);
+
+  @override
+  String toString() => "${value}${unit == null ? '' : ' ' + unit}";
+}
+
+Map<String, Map<String, SensorValue>> parseSensorsOutput(String input) {
+  var out = <String, Map<String, SensorValue>>{};
```

Comment:
this is a bit hard to maintain I think. It'd be great to have a test with some sample input taken from the CLI, and see if the output map corresponds with that you expect?

Reply:
Added an example which parses from static inputs. I might make another PR that extracts other components and tests them along with this code.